### DISCUSSION TITLE
Wazuh - Cisco umbrella integration

### DISCUSSION THREAD

**Discussion by mirceacroitoru** [wazuh/wazuh#13349] [Q&A] [answered]
Hi 

I am trying to get into Wazuh some alerts from Azure AD. I configured everything according to the wazuh documentation however i am getting below 403 error.
App permissions are set correctly ( also tried with full permissions ) but still get the 403 error.
If anyone can give me a hint what to try more...

PM INFO: AZURE Azure Graph starting.
PM INFO: AZURE Graph: Getting authentication token.
PM DEBUG: AZURE Starting new HTTPS connection (1): login.microsoftonline.com:443
PM DEBUG: AZURE https://login.microsoftonline.com:443 "POST /company.com/oauth2/token?api-version=1.0 HTTP/1.1" 200 1482
PM INFO: AZURE Graph: Getting data.
PM INFO: AZURE Graph: The search starts from the date: 2022-05-02T14:31:08.1651491068Z for query: 'activities/audit?api-version=beta'
PM DEBUG: AZURE Starting new HTTPS connection (1): graph.windows.net:443
PM DEBUG: AZURE https://graph.windows.net:443 "GET /company.com/activities/audit?api-version=beta&$filter=activityDate+ge+2022-05-02T14:31:08.1651491068Z HTTP/1.1" 403 168
PM INFO: AZURE Graph: Request status: 403
PM ERROR: AZURE Error: The request for the query could not be made: '403 Client Error: Forbidden for url: https://graph.windows.net/company.com/activities/audit?api-version=beta&$filter=activityDate+ge+2022-05-02T14:31:08.1651491068Z'.
PM INFO: AZURE Graph: End

**Comment by jctello** [comment, marked as answer]
Hello @mirceacroitoru,

The error indicates there's an issue with authentication. Either a bad credential or lack of permissions. 
You may verify the credentials using curl by executing:
```
curl -X POST -d 'grant_type=client_credentials&client_id=[APP_ID]&client_secret=[PASSWORD]&scope=https://graph.microsoft.com/.default' https://login.microsoftonline.com/[TENANT_ID]/oauth2/token
```
One critical step is to grant admin consent to the permissions, I was able to see the same issue when I didn't grant this in this screen:
![image](https://user-images.githubusercontent.com/36071202/169086245-727c8d9c-b7ee-4fa8-b7ab-7e40deda98bf.png)

Finally, I see the body of this question is regarding Azure and the title is regarding Cisco Umbrella.
In the case of Cisco Umbrella the logs can be collected by configuring them to send the logs to their Cisco managed S3 bucket. Information on how to configure this is shown here: https://documentation.wazuh.com/current/amazon/services/supported-services/cisco-umbrella.html

This uses our AWS wodle so the instructions assume you've configured a profile file, however you can also specify the access key and secret key information directly in the configuration file. For more information see here: https://documentation.wazuh.com/current/amazon/services/prerequisites/credentials.html#authenticating-options

A simple configuration example would be:
```
<wodle name="aws-s3">

  <disabled>no</disabled>
  <interval>10m</interval>
  <run_on_start>yes</run_on_start>
  <skip_on_error>yes</skip_on_error>

  <bucket type="cisco_umbrella">
    <name>cisco-managed-us-east-1</name>
    <path>123456_abcdef0123456789/dnslogs</path>
  <access_key>insert_access_key</access_key>
  <secret_key>insert_secret_key</secret_key>
  </bucket>

</wodle>
```
